exec/java-exec/src/main/java/org/apache/drill/exec/server/Drillbit.java

```diff
@@ -83,18 +86,20 @@
   @VisibleForTesting
   public Drillbit(
       final DrillConfig config,
+      final CaseInsensitiveMap<OptionValidator> validators,
       final RemoteServiceSet serviceSet) throws Exception {
-    this(config, serviceSet, ClassPathScanner.fromPrescan(config));
+    this(config, validators, serviceSet, ClassPathScanner.fromPrescan(config));
```

Comment:
Interesting... Why was this change needed?

To be honest, I'm not convinced that the set of inputs is an *input* to the Drillbit rather than an internal *attribute* of the Drillbit. That is, the Drillbit should be responsible for defining its set of valid options, not the code that houses the Drillbit.

Reply:
Currently there are no internal validators defined, so in order to test the logic for internal validators I had to create a set of mock validators which included internal validators, and use those for testing. I have changed this so that the Drillbit constructors which take a set of validators are annotated to be visible for testing.